File: WMS/WMS/BlockHeader.cs
using System;
using Newtonsoft.Json;

namespace WMS
{
    /// <summary>
    /// 分块传输协议头
    /// </summary>
    public class BlockHeader
    {
        /// <summary>
        /// 同步的表名（如Item、User）
        /// </summary>
        [JsonProperty("Table")]
        public string table { get; set; }

        /// <summary>
        /// 总块数
        /// </summary>
        [JsonProperty("TotalBlocks")]
        public int totalBlocks { get; set; }

        /// <summary>
        /// 当前块号（从1开始）
        /// </summary>
        [JsonProperty("CurrentBlock")]
        public int currentBlock { get; set; }

        /// <summary>
        /// 是否为最后一块
        /// </summary>
        [JsonProperty("IsLastBlock")]
        public bool isLastBlock { get; set; }

        /// <summary>
        /// 本块数据条数
        /// </summary>
        [JsonProperty("DataCount")]
        public int dataCount { get; set; }

        /// <summary>
        /// 指令类型（复用你现有Reply/Insert）
        /// </summary>
        [JsonProperty("Command")]
        public string command { get; set; }

        /// <summary>
        /// 操作类型（复用你现有Insert/Select/Sync）
        /// </summary>
        [JsonProperty("Action")]
        public string action { get; set; }
    }
}

File: CTWMS2/WMS/app/src/main/java/com/example/android/wms/Socket/BlockHeader.kt
package com.example.android.wms.Socket

import com.google.gson.annotations.SerializedName
import kotlinx.serialization.InternalSerializationApi
import kotlinx.serialization.Serializable


data class BlockHeader(
        val Table: String,        // 表名（User/Item）
        val TotalBlocks: Int,     // 总块数
        val CurrentBlock: Int,    // 当前块号
        val IsLastBlock: Boolean, // 是否最后一块
        val DataCount: Int,       // 本块数据条数
        val Command: String,      // 指令（Reply）
        val Action: String        // 操作（Insert）
)
//@InternalSerializationApi /**
// * 分块传输协议头（与服务端BlockHeader完全一致）
// */
//@Serializable
//data class BlockHeader(
//        @SerializedName("Table")
//        val table: String, // 表名（Item/User/Printer）
//        @SerializedName("TotalBlocks")
//        val totalBlocks: Int, // 总块数
//        @SerializedName("CurrentBlock")
//        val currentBlock: Int, // 当前块号
//        @SerializedName("IsLastBlock")
//        val isLastBlock: Boolean, // 是否最后一块
//        @SerializedName("DataCount")
//        val dataCount: Int, // 本块数据条数
//        @SerializedName("Command")
//        val command: String, // 指令（Reply）
//        @SerializedName("Action")
//        val action: String // 操作（Insert/Sync）
//)